File: generate_sql.php
<?php
/**
 * Генерация SQL файла из polesie_products.php
 */

$data = require '/workspace/polesie_products.php';

$sql = "-- =====================================================================\n";
$sql .= "-- ПОЛНАЯ НОМЕНКЛАТУРА ПРОДУКЦИИ ОАО «ПОЛЕСЬЕЭЛЕКТРОМАШ»\n";
$sql .= "-- Файл: database_updated.sql\n";
$sql .= "-- Источник: polesie_products.php\n";
$sql .= "-- =====================================================================\n\n";
$sql .= "SET FOREIGN_KEY_CHECKS = 0;\n\n";
$sql .= "-- Очистка существующих данных\n";
$sql .= "DELETE FROM `products`;\n";
$sql .= "DELETE FROM `product_categories`;\n\n";

// Категории
$sql .= "-- =====================================================================\n";
$sql .= "-- КАТЕГОРИИ ПРОДУКЦИИ\n";
$sql .= "-- =====================================================================\n";
$sql .= "INSERT INTO `product_categories` (`id`, `name`, `slug`, `parent_id`) VALUES\n";

$categoryMap = [
    'asynchronous_three_phase_general' => 1,
    'energy_efficient' => 2,
    'multispeed' => 3,
    'high_slip' => 4,
    'for_pumps' => 5,
    'for_gearboxes' => 6,
    'for_poultry' => 7,
    'for_railway' => 8,
    'explosion_proof' => 9,
    'single_phase' => 10,
    'pumps' => 11,
    'other_products' => 12,
];

$categories = [];
$categoryId = 1;
foreach ($data['categories'] as $key => $cat) {
    if (isset($categoryMap[$key])) {
        $id = $categoryMap[$key];
        $name = addslashes($cat['name']);
        $slug = str_replace('_', '-', $key);
        $categories[] = "($id, '$name', '$slug', NULL)";
        $categoryId++;
    }
}
$sql .= implode(",\n", $categories) . ";\n\n";

// Продукция
$productInserts = [];
$productId = 1;

foreach ($data['categories'] as $key => $cat) {
    if (!isset($categoryMap[$key])) continue;
    
    $catId = $categoryMap[$key];
    
    if (isset($cat['models'])) {
        foreach ($cat['models'] as $model) {
            $article = $model['model'];
            $name = 'Двигатель ' . $article;
            
            // Создаем спецификации
            $specs = [];
            foreach ($model as $k => $v) {
                if ($k !== 'model') {
                    $specs[$k] = $v;
                }
            }
            $specsJson = json_encode($specs, JSON_UNESCAPED_UNICODE);
            
            $price = isset($model['мощность_квт']) ? round($model['мощность_квт'] * 100 + 200, 2) : 300.00;
            $image = 'motor_' . strtolower(preg_replace('/[^A-Za-z0-9]/', '_', $article)) . '.jpg';
            
            $productInserts[] = "('$article', '$name', $catId, 1, '$specsJson', '$image', $price, 'BYN', TRUE)";
            $productId++;
        }
    }
    
    // Подкатегории (для насосов и т.д.)
    if (isset($cat['subcategories'])) {
        foreach ($cat['subcategories'] as $subkey => $subcat) {
            if (isset($subcat['parameters'])) {
                $article = isset($subcat['parameters']['потребляемая_мощность_вт']) ? 'БЦ-0,5-20-У1.1' : 'ГНОМ';
                $name = $subcat['name'];
                $specsJson = json_encode($subcat['parameters'], JSON_UNESCAPED_UNICODE);
                $price = isset($subcat['parameters']['потребляемая_мощность_вт']) ? 180.00 : 250.00;
                $image = $subkey === 'centrifugal_household' ? 'pump_bc05.jpg' : 'pump_gnom.jpg';
                
                $productInserts[] = "('$article', '$name', $catId, 1, '$specsJson', '$image', $price, 'BYN', TRUE)";
                $productId++;
            }
        }
    }
    
    // Прочая продукция
    if (isset($cat['items'])) {
        foreach ($cat['items'] as $item) {
            if (isset($item['models'])) {
                foreach ($item['models'] as $model) {
                    $name = $item['name'] . ' ' . $model;
                    $specs = ['назначение' => 'Для бытовых электроплит'];
                    $specsJson = json_encode($specs, JSON_UNESCAPED_UNICODE);
                    $price = 45.00 + (strpos($model, '180') !== false ? 10 : 0) + (strpos($model, '2.0') !== false ? 20 : 0);
                    $image = 'ekch' . preg_replace('/[^0-9]/', '', $model) . '.jpg';
                    
                    $productInserts[] = "('$model', '$name', $catId, 1, '$specsJson', '$image', $price, 'BYN', TRUE)";
                    $productId++;
                }
            }
        }
    }
}

$sql .= "-- =====================================================================\n";
$sql .= "-- ПРОДУКЦИЯ\n";
$sql .= "-- =====================================================================\n";
$sql .= "INSERT INTO `products` (`article`, `name`, `category_id`, `base_unit_id`, `specifications`, `image`, `base_price`, `currency`, `is_active`) VALUES\n";
$sql .= implode(",\n", $productInserts) . ";\n\n";

$sql .= "SET FOREIGN_KEY_CHECKS = 1;\n";

file_put_contents('/workspace/polesie/sql/database_updated.sql', $sql);
echo "SQL файл успешно создан!\n";
echo "Строк: " . count(explode("\n", $sql)) . "\n";
echo "Продуктов: " . count($productInserts) . "\n";
echo "Категорий: " . count($categories) . "\n";

File: generate_sql.py
#!/usr/bin/env python3
import re
import json

# Читаем polesie_products.php и извлекаем данные
with open('/workspace/polesie_products.php', 'r', encoding='utf-8') as f:
    content = f.read()

# Парсим модели двигателей
models_data = {
    'asynchronous_three_phase_general': [],
    'energy_efficient': [],
    'multispeed': [],
    'high_slip': [],
    'for_pumps': [],
    'for_gearboxes': [],
    'for_poultry': [],
    'for_railway': [],
    'explosion_proof': [],
    'single_phase': [],
    'centrifugal_household': [],
    'submersible_dirty': [],
    'other_products': [],
}

# Категория 1: АИР (общепромышленные)
air_pattern = r"\['model'\s*=>\s*'([^']+)',\s*'мощность_квт'\s*=>\s*([0-9.]+),\s*'частота_вращения_об_мин'\s*=>\s*(\d+),\s*'кпд_проц'\s*=>\s*([0-9.]+)"
matches = re.findall(air_pattern, content)

# Разбираем вручную по секциям
lines = content.split('\n')
current_category = None
current_model = {}
in_models_section = False

for i, line in enumerate(lines):
    # Определяем категорию
    if "'asynchronous_three_phase_general'" in line and "=>" in line:
        current_category = 'asynchronous_three_phase_general'
    elif "'energy_efficient'" in line and "=>" in line:
        current_category = 'energy_efficient'
    elif "'multispeed'" in line and "=>" in line:
        current_category = 'multispeed'
    elif "'high_slip'" in line and "=>" in line:
        current_category = 'high_slip'
    elif "'for_pumps'" in line and "=>" in line:
        current_category = 'for_pumps'
    elif "'for_gearboxes'" in line and "=>" in line:
        current_category = 'for_gearboxes'
    elif "'for_poultry'" in line and "=>" in line:
        current_category = 'for_poultry'
    elif "'for_railway'" in line and "=>" in line:
        current_category = 'for_railway'
    elif "'explosion_proof'" in line and "=>" in line:
        current_category = 'explosion_proof'
    elif "'single_phase'" in line and "=>" in line:
        current_category = 'single_phase'
    elif "'centrifugal_household'" in line and "=>" in line:
        current_category = 'centrifugal_household'
    elif "'submersible_dirty'" in line and "=>" in line:
        current_category = 'submersible_dirty'
    elif "'other_products'" in line and "=>" in line:
        current_category = 'other_products'
    
    # Собираем модели
    if current_category and current_category not in ['centrifugal_household', 'submersible_dirty', 'other_products']:
        model_match = re.search(r"'model'\s*=>\s*'([^']+)'", line)
        if model_match:
            if current_model:
                models_data[current_category].append(current_model)
            current_model = {'model': model_match.group(1)}
        
        # Собираем параметры
        if current_model:
            param_match = re.search(r"'([^']+)'?\s*=>\s*([^,\]]+)", line)
            if param_match and param_match.group(1) != 'model':
                key = param_match.group(1).strip()
                val = param_match.group(2).strip().rstrip(',')
                # Очищаем значение
                if val.startswith("'") and val.endswith("'"):
                    val = val[1:-1]
                elif val.startswith('"') and val.endswith('"'):
                    val = val[1:-1]
                else:
                    try:
                        if '.' in val:
                            val = float(val)
                        else:
                            val = int(val)
                    except:
                        pass
                current_model[key] = val

# Добавляем последний модель
if current_model and current_category:
    models_data[current_category].append(current_model)

print("Найдено моделей по категориям:")
for cat, models in models_data.items():
    print(f"  {cat}: {len(models)}")

